core/dbt/task/base.py

```diff
@@ -192,6 +192,20 @@ def __init__(self, node):
         self.node = node
 
 
+def mark_node_as_skipped(node, message):
```

Comment:
Is there anywhere else that was considered to define this method? Having it straggling at the top-level of the base task doesn't feel like a good spot. Its responsibility is in creating a `RunResult` object from a node + message. Perhaps a classmethod on `RunResult`? For example, `RunResult.from_node(node, message, status)`?